src/WebJobs.Script.WebHost/Security/KeyManagement/DefaultKeyValueConverterFactory.cs

```diff
@@ -16,9 +16,9 @@ public sealed class DefaultKeyValueConverterFactory : IKeyValueConverterFactory
         private readonly bool _encryptionSupported;
         private static readonly PlaintextKeyValueConverter PlaintextValueConverter = new PlaintextKeyValueConverter(FileAccess.ReadWrite);
 
-        public DefaultKeyValueConverterFactory()
+        public DefaultKeyValueConverterFactory(bool isEncryptionSupported)
         {
-            _encryptionSupported = IsEncryptionSupported();
+            _encryptionSupported = !isEncryptionSupported && IsEncryptionSupported();
```

Comment:
nit: This flag name is a bit confusing. Perhaps change it to something like `allowEncryption` or something along those lines and add a documentation stating that setting that argument to true will cause the factory to use encryption if supported by the current environment.